sqlmesh/dbt/model.py

```diff
@@ -204,6 +203,19 @@ def _extract_sql_config(self) -> None:
         self._sql_no_config = SqlStr(no_config)
         self._sql_embedded_config = SqlStr(embedded_config)
 
+    @property
+    def _big_query_partition_by_expr(self) -> exp.Expression:
+        assert isinstance(self.partition_by, dict)
+        data_type = self.partition_by["data_type"].lower()
+        if data_type == "int64" or (
+            data_type == "date" and self.partition_by["granularity"].lower() == "day"
+        ):
+            return exp.to_column(self.partition_by["field"])
+        return d.parse_one(
```

Comment:
let's not do the parse, just choose the appropriate expression, timestamptrunc or datetrunc,